packages/gatsby-mdx/gatsby-node.js

```diff
@@ -22,27 +32,65 @@ exports.onCreatePage = require("./gatsby/on-create-page");
  */
 exports.onCreateWebpackConfig = require("./gatsby/create-webpack-config");
 
+/**
+ * Create the cache directories
+ */
+
+exports.onPreBootstrap = () => {
```

Comment:
Create the cache directories at the start instead of every time we cache something.